aws/lambda-with-other-services/lambda-dynamodb/lambda-dynamodb.md

```diff
@@ -1,3 +1,298 @@
 # Writing to a DynamoDB Table Using a Lambda Function
 
 ### By Andrew Levy
+
+In this tutorial we will demonstrate how to use an AWS Lambda function to transfer a file from one S3 bucket to another.
+
+## The Table
+
+For this example we will use a table whose primary key is a unique string. Once the table has been created, copy down its Amazon Resource Name (ARN) as that will be used to give the Lambda function access to the table.
+
+![alt text](images/1.png)
+
+![alt text](images/2.png)
+
+## Permissions For Lambda
+
+We will need to give our function permission to write to our table. We do this by giving the function a role. Your technical supervisor may provide you a role, but if not you will need to create one. However, even if you are not creating the role yourself, it is still important to understand how roles and permissions work in AWS.
+
+To create the role, go to the console for the AWS IAM service. Select the "Policies" tab, and then click the "Create policy" button.
+
+![alt text](images/3.png)
+
+In keeping with best practices, we will give the function the fewest permissions necessary to complete its intended task. It will only be able to write to DynamoDB (via the PutItem function), and only the one table that we specify.
+
+Select the JSON tab to write out the policy. This JSON code will provide the necessary access:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:PutItem"
+            ],
+            "Resource":[
+		"Table’s ARN goes here"
+            ]
+        }
+    ]
+}
+```
+
+![alt text](images/4.png)
+
+Click "Review Policy," give the policy a name and description, and click "Create Policy."
```

Comment:
This part is less necessary. It's already covered [here](https://github.com/mkdecisiondev/devdocs/blob/backended/aws/introduction-to-aws/iam/iam.md#creating-a-role-and-adding-policies-to-it). Read that section before continuing. Perhaps you can show the policy code and image, and then say something like, "If you've forgotten how to create policies, you can read the [IAM section](aws/introduction-to-aws/iam/iam.md#creating-a-role-and-adding-policies-to-it) on how to do so." Put the link in there while you're at it, it will be "aws/introduction-to-aws/iam/iam.md#creating-a-role-and-adding-policies-to-it".